services/core-web/src/components/admin/complianceCodes/ComplianceReportManagement.tsx

```diff
@@ -14,191 +20,205 @@ import { removeNullValues } from "@mds/common/constants/utils";
-                        gutter={[0, 16]}
-                    >
-                        <Input
-                            placeholder="Search Compliance Article"
-                            onChange={(e) => setSectionSearchText(e.target.value)}
-                            onPressEnter={() => handleSectionSearch(confirm, sectionSearchText)}
-                        />
-                        <Button
-                            onClick={() => handleSectionSearch(confirm, sectionSearchText)}
-                            icon={<SearchOutlined />}
-                            size="small"
-                            type="primary"
-                        >Search</Button>
-                        <Button
-                            onClick={() => {
-                                setSectionSearchText(null);
-                                handleSectionSearch(confirm, null);
-                            }}
-                            size="small"
-                        >
-                            Reset
-                        </Button>
-                    </Row>
-                </div>
-            );
-        }
-    }
-    const columns = [
-        renderTextColumn("report_name", "Report Name", true),
-        {
-            ...renderTextColumn("section", "Section"),
-            sorter: true,
-            ...sectionFilter
-        },
-        {
-            ...renderTextColumn("is_prr_only", "Report Type"),
-            filters: [
-                { value: false, text: "Code Required Report" },
-                { value: true, text: "Permit Required Report" }
-            ]
-        },
-        {
-            ...renderTextColumn("regulatory_authority", "Regulatory Authority"),
-            filters: regulatoryAuthorityFilter,
-            sorter: true
-        },
-        renderActionsColumn({ actions })
-    ];
-
-    const formatCode = (article: IComplianceArticle) => {
-        if (!article) {
-            return {
-                regulatory_authority: EMPTY_FIELD,
-                section: EMPTY_FIELD
-            }
-        }
-        return {
-            regulatory_authority: REPORT_REGULATORY_AUTHORITY_ENUM[article.cim_or_cpo] ?? REPORT_REGULATORY_AUTHORITY_CODES.NONE,
-            section: formatComplianceCodeArticleNumber(article)
-        }
+  const dispatch = useDispatch();
+  const { search } = useLocation();
+  const history = useHistory();
+  const initialParams = queryString.parse(search);
+  const reportPageData = useSelector(getComplianceReportPageData);
+  const reportDefinitions = reportPageData?.records ?? [];
+  const [sectionSearchText, setSectionSearchText] = useState(null);
+  const [queryParams, setQueryParams] = useState<ComplianceReportParams>({ ...defaultParams, ...initialParams });
+  const isLoaded = useSelector(getReportDefinitionsLoaded(queryParams));
+
+  const fetchData = () => {
+    dispatch(fetchComplianceReports(queryParams));
+    dispatch(fetchMineReportDueDateTypes({}));
```

Comment:
might actually be nice to separate this call out from fetchData. fetchData is called every time that the table sort/filter/pagination is changed. But I don't think the due date types change with any kind of frequency like that. Could make sense to put it in its own useEffect to only call it once on initial render

Reply:
that or have a loading indicator so that it tests if it's loaded in the state before fetching it